spec/unit/libraries/recipe_spec.rb

```diff
@@ -47,4 +48,93 @@
       end
     end
   end
+
+  describe '.my_cluster_data' do
+    before do
+      allow(ChefVault::Item).to receive(:data_bag_item_type).and_return(:vault)
+      allow(ChefVault::Item).to receive(:load).with('cerner_splunk', 'cluster_bag').and_return(cluster_configs)
+      allow(ChefVault::Item).to receive(:load).with('cerner_splunk', 'multisite_bag').and_return(multisite_configs)
+    end
+
+    after do
+      CernerSplunk.reset
+    end
+
+    let(:clusters) { ['cerner_splunk/cluster_bag'] }
+    let(:node) { { 'splunk' => { 'config' => { 'clusters' => clusters } } } }
+    let(:cluster_configs) { { 'site' => 'site1', 'multisite' => 'cerner_splunk/multisite_bag', 'apps' => 'cerner_splunk/apps', 'indexes' => 'cerner_splunk/overriden-indexes' } }
+    let(:multisite_configs) { { 'sites' => ['cerner_splunk/cluster_bag'], 'master_uri' => 'https://33.33.33.11:8089', 'indexes' => 'cerner_splunk/indexes' } }
+
+    subject { CernerSplunk.my_cluster_data(node) }
+
+    context 'cluster is a multisite cluster' do
+      it 'merges multisite and cluster attributes' do
+        expected_attributes = {
+          'sites' => ['cerner_splunk/cluster_bag'],
+          'master_uri' => 'https://33.33.33.11:8089',
+          'site' => 'site1',
+          'multisite' => 'cerner_splunk/multisite_bag',
+          'indexes' => 'cerner_splunk/overriden-indexes',
+          'apps' => 'cerner_splunk/apps'
+        }
+        expect(subject).to eq(expected_attributes)
+      end
+    end
+
+    context 'cluster is a singlesite cluster' do
+      let(:cluster_configs) { { 'master_uri' => 'https://33.33.33.11:8089', 'replication_ports' => { '8080' => {} }, 'receivers' => ['33.33.33.33'] } }
+
+      it 'merges multisite and cluster attributes' do
```

Comment:
it 'only returns the cluster config', or it 'does not merge the multisite cluster data', or something along those lines.

Reply:
0cd73923d94c8e5f81dede79457fbf21d959f065